jax/numpy/lax_numpy.py

```diff
@@ -2442,6 +2442,82 @@ def cov(m, y=None, rowvar=True, bias=False, ddof=None, fweights=None,
   return true_divide(dot(X, X_T.conj()), f).squeeze()
 
 
+@_wraps(onp.quantile)
+def quantile(a, q, axis=None, out=None, overwrite_input=False,
+             interpolation="linear", keepdims=False):
+  if overwrite_input or out is not None:
+    msg = ("jax.numpy.quantile does not support overwrite_input=True or "
+           "out != None")
+    raise ValueError(msg)
+  if interpolation != "linear":
+    raise NotImplementedError("Only interpolation='linear' is implemented")
+
+  a = asarray(a)
```

Comment:
I tend to skip these `asarray` calls because I think they can be slow. I'm not necessarily advocating for that though because I'm fuzzy on this.

Should our policy be to call `asarray` as much as NumPy itself does? Do you have a sense for this?

Reply:
I personally find it useful ergonomically to allow lists here.

We previously discussed this in other contexts and reached a compromise, which was to add an internal version of `asarray` that does not accept tracers inside Python data structures. When it exists we should use it here.